### PR TITLE
Add documentation and tests, refactor code, fix edge cases

### PR DESCRIPTION
**Pull request by tmaklin**

### CODE REVIEW

File: src/derandomize.rs [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+//! Derandomizing noisy _k_-bounded matching statistics.
+
+/// Evaluates the CDF of _k_-bounded matching statistics random match distribution.
+///
+/// Computes the log-probability that a matching statistic with value
+/// `t` or less that is the result of mapping a _k_-mer with
+/// `alphabet_size` possible characters against an index containing
+/// `n_kmers` _k_-mers was generated by chance.
+///
+/// TODO Add formula to log_rm_max_cdf documentation
```
Comment (tmaklin) [inline src/derandomize.rs:23]: A new Todo was discovered. If it is not a priority right now,consider marking it for later attention.
TODO Add formula to log_rm_max_cdf documentation
- [ ] Ignore

Comment (tmaklin) [inline src/derandomize.rs:23]: A new Todo was discovered. If it is not a priority right now,consider marking it for later attention.
TODO Add formula to log_rm_max_cdf documentation
- [ ] Ignore

Comment (tmaklin) [inline src/derandomize.rs:23]: A new Todo was discovered. If it is not a priority right now,consider marking it for later attention.
TODO Add formula to log_rm_max_cdf documentation
- [ ] Ignore
